include/EVT/utils/time.hpp

```diff
@@ -3,13 +3,17 @@
  * operations are platform independent.
  *
  */
-
 #ifndef _EVT_TIME_
 #define _EVT_TIME_
 
 #include <cstdint>
 
-#include <HALf3/stm32f3xx_hal.h>
+#ifdef STM32f4xx
```

Comment:
Instead of doing this, you should break time apart into timef3xx and tim4xx like we have for all the IO stuff